### DISCUSSION TITLE
PCIe TPU?

### DISCUSSION THREAD

**Discussion by Cheesnachos** [geezacoleman/OpenWeedLocator#130] [Ideas]
I ran across the following [hat from pineboards](https://pineboards.io/products/hat-mpcie-coral-edge-tpu-bundle-for-raspberry-pi-5). Should also work with OWL?

**Comment by dpicassom** [comment]
Raspberry Pi Just released an official AI HAT, 13 TOPS:
[Raspberry Pi AI Kit available now at $70](https://www.raspberrypi.com/news/raspberry-pi-ai-kit-available-now-at-70/)

Might help with differentiating crop from cover crop and weed

**Comment by geezacoleman** [comment]
I did order this hardware (unfortunately haven't had a chance to try integrating it yet) and it should work in theory. Pineboards have also provided [good documentation](https://pineboards.io/blogs/tutorials/how-to-configure-the-google-coral-edge-tpu-on-the-raspberry-pi-5) on getting it working. The biggest challenge it seems is now the largely unsupported pycoral library, and a worrying lack of Google support for the Coral project altogether.

![image](https://github.com/geezacoleman/OpenWeedLocator/assets/51358498/1161acd8-b852-4823-80f3-22c06c11e3ae)

Plus, there are issues with image size and throughput of the Coral TPU, well documented on this [youtube explainer](https://www.youtube.com/watch?v=HMeDLSflCA0). There are some other [good explainers](https://www.youtube.com/watch?v=sOxQTRRh9tw) on this too, where you can achieve relatively high ~30FPS framerates with low resolution images.

So all in all, well worth exploring, but I think we will begin to see better supported hardware and software emerge.
